Description: Don't fail with Numpy 1.19

Creating a nested DataFrame (which was already not recommended)
via the constructor no longer works.
Give a clearer error and xfail the tests.

Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>
Bug: https://github.com/pandas-dev/pandas/issues/32289
Forwarded: no

--- pandas-1.0.4+dfsg.orig/pandas/core/internals/construction.py
+++ pandas-1.0.4+dfsg/pandas/core/internals/construction.py
@@ -292,7 +292,7 @@ def prep_ndarray(values, copy=True) -> n
     if values.ndim == 1:
         values = values.reshape((values.shape[0], 1))
     elif values.ndim != 2:
-        raise ValueError("Must pass 2-d input")
+        raise ValueError("Plain DataFrames must be 2-d - for higher dimensions use MultiIndex or the python3-xarray package.  If you are trying to create a nested DataFrame (which is not recommended) see https://github.com/pandas-dev/pandas/issues/32289")
 
     return values
 
--- pandas-1.0.4+dfsg.orig/pandas/tests/frame/test_constructors.py
+++ pandas-1.0.4+dfsg/pandas/tests/frame/test_constructors.py
@@ -145,6 +145,7 @@ class TestDataFrameConstructors:
         assert df.loc[1, 0] is None
         assert df.loc[0, 1] == "2"
 
+    @pytest.mark.xfail(condition=True,strict=False,raises=ValueError, reason="Interprets list of frame as 3D, https://github.com/pandas-dev/pandas/issues/32289")
     def test_constructor_list_frames(self):
         # see gh-3243
         result = DataFrame([DataFrame()])
@@ -477,7 +477,7 @@ class TestDataFrameConstructors:
             DataFrame(index=[0], columns=range(0, 4), data=arr)
 
         # higher dim raise exception
-        with pytest.raises(ValueError, match="Must pass 2-d input"):
+        with pytest.raises(ValueError, match="Plain DataFrames must be 2-d"):
             DataFrame(np.zeros((3, 3, 3)), columns=["A", "B", "C"], index=[1])
 
         # wrong size axis labels
@@ -498,6 +498,7 @@ class TestDataFrameConstructors:
         with pytest.raises(ValueError, match=msg):
             DataFrame({"a": False, "b": True})
 
+    @pytest.mark.xfail(condition=True,strict=False,raises=ValueError, reason="Interprets embedded frame as 3D, https://github.com/pandas-dev/pandas/issues/32289")
     def test_constructor_with_embedded_frames(self):
 
         # embedded data frames
@@ -764,7 +764,7 @@ class TestDataFrameConstructors:
             DataFrame(mat, columns=["A", "B"], index=[1, 2])
 
         # higher dim raise exception
-        with pytest.raises(ValueError, match="Must pass 2-d input"):
+        with pytest.raises(ValueError, match="Plain DataFrames must be 2-d"):
             DataFrame(empty((3, 3, 3)), columns=["A", "B", "C"], index=[1])
 
         # automatic labeling
